### DISCUSSION TITLE
Add support for username based ACL for Redis and also have configurable database id

### DISCUSSION THREAD

**Discussion by Veera93** [Netflix/conductor#3713] [Ideas]
In the current version of Orkes, the Redis configuration only allows for password-based authentication. However, starting from Redis version 6.0.0, support for ACL (Access Control List) with both username and password was introduced. Can we consider adding username support to the configuration?

In addition to this, currently in the configuration the code connects to the default database ID(0). Can we have this configurable as well? 

Describe the solution you'd like
Jedis library already has support for username and password-based configuration. We can wire the username in the properties and default it to null if the username is not configured. The same thing can be done for database id as well.

I can create the PR, if I get the approval for this issue.

**Comment by Anupchat** [comment]
Added PR here can you take a look - https://github.com/Netflix/conductor/pull/3847

**Reply by Anupchat** [reply]
@apanicker-nflx , @aravindanr , @manan164 , @v1r3n can you take a look into this and review the PR above. We would like this for the Redis infrastructure that is in use at VMware.
